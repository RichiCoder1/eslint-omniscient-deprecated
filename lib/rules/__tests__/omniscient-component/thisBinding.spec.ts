import { fix, withRuleOptions } from "./__setup__/setup";

describe("this binding", () => {
    it("should handle this in render func", () => {
        const input = `
import component from 'omniscient';

const TestComponent = component("TestComponent", {
    getInitialState() {
        return { i: 1 };
    },
    handleClick(e) {
        e.preventDefault();
    },
    doThing() {
        console.log("asdasd");
    },
    thatThing(a) {
        return a;
    }
}, ({label}) => {
    const options = {
        onThing: this.doThing,
        thatThing: this.thatThing(1)
    };
    return <button onClick={this.handleClick}>Test</button>;
})`;
        const report = fix(input);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });

    it("should handle this in render func without class props", () => {
        const input = `
import component from 'omniscient';

const TestComponent = component("TestComponent", {
    getInitialState() {
        return { i: 1 };
    },
    handleClick(e) {
        e.preventDefault();
    },
    doThing() {
        console.log("asdasd");
    },
    thatThing(a) {
        return a;
    }
}, ({label}) => {
    const options = {
        onThing: this.doThing,
        thatThing: this.thatThing(1)
    };
    return <button onClick={this.handleClick}>Test</button>;
})`;
        const report = fix(
            input,
            withRuleOptions({
                useClassProperties: false,
            })
        );
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });
});
