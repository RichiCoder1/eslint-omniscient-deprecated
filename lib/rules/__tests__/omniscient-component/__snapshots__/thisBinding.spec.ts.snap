// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`this binding should handle this in render func 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class TestComponent extends Component {
    static state = { i: 1 };

    static displayName = \\"TestComponent\\";

    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }
    thatThing = a => {
        return a;
    };

    doThing = () => {
        console.log(\\"asdasd\\");
    };

    handleClick = e => {
        e.preventDefault();
    };

    render() {
        const { label } = this.props;
        const options = {
            onThing: this.doThing,
            thatThing: this.thatThing(1)
        };
        return <button onClick={this.handleClick}>Test</button>;
    }
}",
}
`;

exports[`this binding should handle this in render func without class props 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class TestComponent extends Component {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }
    thatThing(a) {
        return a;
    }

    doThing() {
        console.log(\\"asdasd\\");
    }

    handleClick(e) {
        e.preventDefault();
    }

    render() {
        const { label } = this.props;
        const options = {
            onThing: this.doThing.bind(this),
            thatThing: this.thatThing(1)
        };
        return <button onClick={this.handleClick.bind(this)}>Test</button>;
    }
}

TestComponent.displayName = \\"TestComponent\\";
TestComponent.state = { i: 1 };",
}
`;
